File: backend/app/routers/audit.py
"""
Audit log endpoints - read-only, admin/analyst visibility.
"""
from typing import Optional
from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from app.core.database import get_db
from app.core.security import require_role
from app.models.audit_log import AuditLog
from app.models.user import User
from app.schemas import AuditLogOut

router = APIRouter(prefix="/api/audit-logs", tags=["audit"])


@router.get("", response_model=list[AuditLogOut])
def list_audit_logs(
    action: Optional[str] = Query(None),
    target_type: Optional[str] = Query(None),
    limit: int = Query(100, le=500),
    offset: int = Query(0, ge=0),
    db: Session = Depends(get_db),
    current_user: User = Depends(require_role("admin", "analyst")),
):
    q = db.query(AuditLog)
    if action:
        q = q.filter(AuditLog.action == action)
    if target_type:
        q = q.filter(AuditLog.target_type == target_type)

    return q.order_by(AuditLog.created_at.desc()).offset(offset).limit(limit).all()

File: frontend/src/api/audit.js
import api from "./client";

export async function listAuditLogs(filters = {}) {
  const params = {};
  if (filters.action) params.action = filters.action;
  if (filters.target_type) params.target_type = filters.target_type;
  if (filters.target_id) params.target_id = filters.target_id;
  if (filters.limit) params.limit = filters.limit;
  if (filters.offset) params.offset = filters.offset;

  const { data } = await api.get("/api/audit-logs", { params });
  return data;
}
